symmetric.go

```diff
@@ -273,26 +273,29 @@ func (c *Context) GenerateSecretKeyWithLabel(id, label []byte, bits int, cipher
 		return nil, errClosed
 	}
 
-	if err := notNilBytes(id, "id"); err != nil {
-		return nil, err
-	}
-	if err := notNilBytes(label, "label"); err != nil {
+	template, err := NewAttributewithLabel(id, label)
+	if err != nil {
 		return nil, err
 	}
-
-	return c.generateSecretKey(id, label, bits, cipher)
+	return c.GenerateSecretKeyWithAttributes(template, bits, cipher)
 
 }
 
-// generateSecretKey creates an secret key of given length and type.
-func (c *Context) generateSecretKey(id, label []byte, bits int, cipher *SymmetricCipher) (k *SecretKey, err error) {
+// GenerateSecretKeyWithAttributes creates an secret key of given length and type. Additional attributes from public and
+// private will be merged into the default templates used when generating keys. Where conflicts occur, the user
+// supplied attributes will win.
+func (c *Context) GenerateSecretKeyWithAttributes(template *AttributeSet, bits int, cipher *SymmetricCipher) (k *SecretKey, err error) {
+	if err = attributeHasID(template, "secret"); err != nil {
```

Comment:
Per the comments above (in `attributes.go`), I think this check ought to be removed.

Reply:
Done